jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -184,6 +174,48 @@ public void onClick(View view) {
         });
     }
 
+    /** Sends the user entered message if permission was granted. Requests the permission if the app
+     * does not already have it. Also shows a dialog if now message was entered. */
+    private void sendMessage() {
+
+        message = editMessage.getText().toString();
+
+        if (message.length() <= 0) {
+            EnterMessageDialogFragment dialog = new EnterMessageDialogFragment();
+            dialog.show(getFragmentManager(), ENTER_MESSAGE_FRAGMENT_TAG);
+            return;
+        }
+
+        delegate.sendMessage(message);
+        editMessage.setText("");
+
+        if (phoneNumberHasNotBeenEntered()) {
```

Comment:
`phoneNumberHasBeenEntered` is preferred. We get the ‘not’ from `!`

Consider the two options:
a) if phone number has been entered
b) if not phone number has been entered
vs
a) if not phone number has not has been entered
b) if phone number not has been entered

Which pair is the easiest to read?